include/compute/boundary_values/impl/boundary_medium_container.tpp

```diff
@@ -20,7 +20,7 @@ specfem::compute::impl::boundary_medium_container<DimensionType, MediumType,
   const int nx = mesh.ngllx;
 
   for (int ispec = 0; ispec < nspec; ispec++) {
-    if (properties.h_element_types(ispec) == MediumType &&
```

Comment:
Can you add boundary_tags to element_types? Then we can call method `get_element_indices(medium_tag, boundary_tag)` here. We also would not the dependency the properties and boundary dependency here for boundary_values.